### ISSUE TITLE
Allow process to exit with successful exit status when budget fails

### ISSUE THREAD

**Issue by julrich** [sitespeedio/sitespeed.io#1934]
Like described in Slack: It would be nice to have the option to suppress the exit code when running a scan with associated budget, that fails. For some CI setups (we use Bamboo) the test result data (`junit.xml` for us) won't be parsed if the job failed.
In those cases, the job would fail anyway due to the test result parsing.

Maybe adding an option like `--budget.dontFailProcess`, set to `false` by default, would be a solution?

Potential places this probably should be added in the documentation:
https://www.sitespeed.io/documentation/sitespeed.io/continuous-integration/
https://www.sitespeed.io/documentation/sitespeed.io/configuration/
https://www.sitespeed.io/documentation/sitespeed.io/performance-budget/

I would offer to propose a PR for the documentation, I could add some example for why this would be useful / exists with Bamboo as an example. I would also add a Bamboo section to the Continuous Integration page.

**Comment by soulgalore** [comment]
Yep I agree we should fix it. I checked the cli and we still have a hack (since 4.4) for taking care of when we had only --budget as a parameter, that makes it a little more work. We should remove that hack on the next major :)
